include/wil/win32_helpers.h

```diff
@@ -56,45 +71,53 @@ namespace wil
 
     namespace filetime
     {
-        constexpr unsigned long long to_int64(const FILETIME &ft)
+        constexpr unsigned long long to_int64(const FILETIME &ft) WI_NOEXCEPT
         {
+#ifdef __cpp_lib_bit_cast
+            return std::bit_cast<unsigned long long>(ft);
+#else
             // Cannot reinterpret_cast FILETIME* to unsigned long long*
             // due to alignment differences.
             return (static_cast<unsigned long long>(ft.dwHighDateTime) << 32) + ft.dwLowDateTime;
+#endif
         }
 
-        inline FILETIME from_int64(unsigned long long i64)
+        WI_CONSTEXPR_BIT_CAST FILETIME from_int64(unsigned long long i64) WI_NOEXCEPT

```

Comment:
Seems like this could pretty easily be split into two definitions to avoid the need for the macro and keep things relatively clean. e.g.
```c++
#if __cpp_lib_bit_cast >= 201806L
constexpr FILETIME from_int64(unsigned long long i64) WI_NOEXCEPT
{
    // New definition
}
#else
inline FILETIME from_int64(unsigned long long i64) WI_NOEXCEPT
{
    // Old definition
}
#endif
```

Reply:
The `add` function right below this one could be constexpr if the macro is kept.